README.md

```diff
@@ -106,50 +106,92 @@ correcting any deficencies that may exist. This is easily done like so:
-applied according to playbooks in per variant directories under
-live-build/variants. A specific variant can be built by passing in the
-variant's name:
+Each variant will have ansible roles applied according to playbooks in
+per variant directories under live-build/variants. An appliance can be
+built by invoking the gradle task for the variant and platform desired.
+The task name has the form 'build\<Variant\>\<Platform\>'. For instance,
+the task to build the 'internal-minimal' variant for KVM is
+'buildInternalMinimalKvm':
 
-    $ ./scripts/docker-run.sh make internal-minimal
+    $ ./scripts/docker-run.sh gradle buildInternalMinimalKvm
 
 When this completes, the newly built VM artifacts will be contained in
-the "live-build/artifacts" directory:
+the "live-build/build/artifacts/" directory:
 
-    $ ls -l live-build/artifacts
-    total 6.0G
-    -rw-r--r-- 1 root root  975M Apr 30 19:47 internal-minimal.ova
-    -rw-r--r-- 1 root root 1009M Apr 30 19:43 internal-minimal.qcow2
-    -rw-r--r-- 1 root root  2.8G Apr 30 19:44 internal-minimal.vhdx
-    -rw-r--r-- 1 root root  975M Apr 30 19:47 internal-minimal.vmdk
+    $ ls -lh live-build/build/artifacts/
+    total 1.9G
+    -rw-r--r-- 1 root root 275M Jan 11 22:31 internal-minimal-kvm.debs.tar.gz
+    -rw-r--r-- 1 root root   45 Jan 11 22:31 internal-minimal-kvm.migration.tar.gz
+    -rw-r--r-- 1 root root 636M Jan 11 22:33 internal-minimal-kvm.qcow2
+
+The appliance produced will contain a kernel optimized for the
+specified platform (which can be one of 'aws', 'azure', 'esx', 'gcp',
+or 'kvm'). The appliance will also contain kernel modules built for
+that optimized kernel, and perhaps some other modules relevant to that
+platform only.
 
 ### Step 5: Use QEMU for Boot Verfication
 
 Once the live-build artifacts have been generated, we can then leverage
 the "qemu" tool to test the "qcow2" artifact:
 
     $ sudo qemu-system-x86_64 -nographic -m 1G \
-    > -drive file=live-build/artifacts/internal-minimal.qcow2
+    > -drive file=live-build/build/artifacts/internal-minimal-kvm.qcow2
 
 This will attempt to boot the "qcow2" VM image, minimally verifying that
 any changes to the build don't cause a boot failure. Further, after the
 image boots (assuming it boots successfully), one can log in via the
-console and perform any post-boot verification that's required (e.g.
-verify certain packages are installed, etc).
+console (username and password are both 'delphix') and perform any
+post-boot verification that's required (e.g. verify certain packages are
+installed, etc).
 
 To exit "qemu", one can use "Ctrl-A X".
 
+## Building an Upgrade Image
+
+An upgrade image for a particular variant can be built by running the
+'buildUpgradeImage\<Variant\>' tasks. For instance, the task to build
+an upgrade image for the internal-minimal variant is
+'buildUpgradeImageInternalMinimal':
+
+    $ PLATFORMS='kvm aws' ./scripts/docker-run.sh gradle buildUpgradeImageInternalMinimal
```

Comment:
I can certainly add something to `version.info`. At that point the image will be unpacked though, so maybe we could also just query to see which versions of `delphix-entire-` are in the apt repo. Either way though, I agree it would be good to have an explicit check for this failure mode so that we can give users a good error message explaining why they have hit this error and how they should proceed.

> I presume that we do this to allow faster builds in some circumstances, but we'll have to weigh that convenience with the possibility that we might accidentally build images for the wrong platforms

Faster builds and smaller upgrade images. Yes, there certainly is a trade off, but I think it's necessary for now unless we want to more or less give up on running the build outside of the jenkins job; running live build five times in a row to build the uprade image for a single variant is just too painful, IMO.